server/src/main/java/io/crate/role/CreateRoleRequest.java

```diff
@@ -74,5 +93,8 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(isUser);
         }
         out.writeOptionalWriteable(secureHash);
+        if (out.getVersion().onOrAfter(Version.V_5_7_0)) {
+            out.writeMap(jwtProperties);
```

Comment:
Please use type safe streaming and not `writeMap` which uses 1 extra byte for the map implementation and 1 extra byte per each map value.

Reply:
Since streaming type safe map requires explicit writing of each key/value in the predefined order, I added a record as it looked like same effort but stricter and more specific. 

`JwtProperties` is very similar to `SecureHash` with the difference that `fromXContent` utilizes `parser.map` for simplicity which should have same effect as 2 nested loops in `SecureHash`.

Reply:
BTW, order of the properties is important as it will be used on the signature verification and it's order sensitive.
Record helps to maintain order. 

Bit verbose but a local draft with `LinkedHashMap` + type safe writeKey/writeValue with explicit order and types was not any better.